app/scripts/periodDirective.js

```diff
@@ -31,12 +28,15 @@ angular.module('angularFormMessagesExample')
         });
 
         // Map UI onto $viewValue
-        $scope.$watch('[from, to]', function (newVal) {
+        $scope.$watch('[from, to]', function (newVal, oldVal) {
+          if (angular.equals(newVal, oldVal)) { return; }
```

Comment:
Volgens mij niet, want het zijn 2 bij een watch in elkaar gekluste objecten, die dus nooit strict equal zijn.